repository/src/main/java/org/apache/atlas/repository/audit/ESBasedAuditRepository.java

```diff
@@ -328,24 +328,25 @@ private boolean createAuditIndex() throws IOException {
         return isSuccess(response);
     }
 
-    private boolean isFieldLimitDifferent() {
-        JsonNode fieldLimit;
+    private boolean shouldUpdateFieldLimitSetting() {
+        JsonNode currentFieldLimit;
         try {
-            fieldLimit = getIndexFieldLimit();
+            currentFieldLimit = getIndexFieldLimit();
         } catch (IOException e) {
             LOG.error("Problem while retrieving the index field limit!", e);
             return false;
         }
         Integer fieldLimitFromConfigurationFile = configuration.getInt(TOTAL_FIELD_LIMIT);
```

Comment:
@nikhilbonte21 Is it possible that TOTAL_FIELD_LIMIT is not defined in configuration? If that happens, what would the value for `fieldLimitFromConfigurationFile` be?

Reply:
Updated logic to get 0 as default in case property is not defined

Reply:
If not defined, it will throw java.util.NoSuchElementException